bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableEnvironment.java

```diff
@@ -0,0 +1,226 @@
+     */
+    void setSystemProperty(final String key, final String value) {
+        if (ignoredProperties.contains(key)) {
+            logDebug("Ignoring system property %s.", key);
+        } else {
+            propertyUpdater.setProperty(key, value);
+        }
+    }
+
+    private static Collection<String> init(final Path jbossHome, final PropertyUpdater propertyUpdater) {
+        final Collection<String> propertyNames = new ArrayList<>();
+        propertyNames.add("java.ext.dirs");
+        propertyNames.add("java.home");
+        propertyNames.add("java.io.tmpdir");
+        propertyNames.add("jboss.server.persist.config");
+        propertyNames.add("jboss.server.management.uuid");
+        propertyNames.add("modules.path");
+        propertyNames.add("user.dir");
+        propertyNames.add("user.home");
+
+        // Configure known paths
+        setSystemProperty(propertyUpdater, "jboss.home.dir", jbossHome, propertyNames);
+        final Path serverBaseDir = resolvePath(jbossHome, "standalone");
+        setSystemProperty(propertyUpdater, "jboss.server.base.dir", serverBaseDir, propertyNames);
+        setSystemProperty(propertyUpdater, "jboss.controller.temp.dir", resolvePath(serverBaseDir, "tmp"), propertyNames);
+        final Path dataDir = resolvePath(serverBaseDir, "data");
+        setSystemProperty(propertyUpdater, "jboss.server.data.dir", dataDir, propertyNames);
+        setSystemProperty(propertyUpdater, "jboss.server.config.dir", resolvePath(serverBaseDir, "config"), propertyNames);
+        setSystemProperty(propertyUpdater, "jboss.server.deploy.dir", resolvePath(dataDir, "content"), propertyNames);
+        setSystemProperty(propertyUpdater, "jboss.server.log.dir", resolvePath(serverBaseDir, "log"), propertyNames);
+        setSystemProperty(propertyUpdater, "jboss.server.temp.dir", resolvePath(serverBaseDir, "tmp"), propertyNames);
+        return propertyNames;
+    }
+
+    private static Path resolvePath(final Path base, final String path1, final String... paths) {
+        Path result = base.resolve(path1);
+        for (String path : paths) {
+            result = result.resolve(path);
+        }
+        return result.toAbsolutePath().normalize();
+    }
+
+    private static void setSystemProperty(final PropertyUpdater propertyUpdater, final String key, final Path path, final Collection<String> names) {
+        names.add(key);
+        final String previousValue = propertyUpdater.setProperty(key, path.toString());
+        if (previousValue == null) {
+            logDebug("Setting system property %s to %s", key, path);
+        } else {
+            logDebug("Replacing system property %s with a value of %s. The previous value was %s.", key, path, previousValue);
+        }
+    }
+
+    @SuppressWarnings("UseOfSystemOutOrSystemErr")
+    static void logDebug(final String format, final Object... args) {
+        if (DEBUG.get()) {
+            System.out.printf("[DEBUG] " + format, args);
```

Comment:
Note this is just an idea. I'm happy to remove it, but felt it may be useful to see some information about the environment when it's booting.